cogboard-app/src/main/kotlin/com/cognifide/cogboard/ssh/SSHClient.kt

```diff
@@ -0,0 +1,181 @@
+package com.cognifide.cogboard.ssh
+
+import com.cognifide.cogboard.CogboardConstants
+import com.cognifide.cogboard.ssh.auth.AuthenticationType
+import com.jcraft.jsch.ChannelExec
+import com.jcraft.jsch.JSch
+import com.jcraft.jsch.JSchException
+import com.jcraft.jsch.Session
+import io.vertx.core.AbstractVerticle
+import io.vertx.core.buffer.Buffer
+import io.vertx.core.eventbus.MessageConsumer
+import io.vertx.core.json.Json
+import io.vertx.core.json.JsonArray
+import io.vertx.core.json.JsonObject
+import java.io.InputStream
+
+class SSHClient : AbstractVerticle() {
+    private lateinit var session: Session
+    private lateinit var jsch: JSch
+    private lateinit var channel: ChannelExec
+    private lateinit var sshInputStream: InputStream
+    private lateinit var consumer: MessageConsumer<JsonObject>
+    override fun start() {
+        registerSSHCommand()
+        super.start()
+    }
+
+    override fun stop() {
+        consumer.unregister()
+        super.stop()
+    }
+
+    private fun registerSSHCommand() {
+        consumer = vertx.eventBus()
+                .consumer<JsonObject>(CogboardConstants.Event.SSH_COMMAND)
+                .handler { message ->
+                    message.body()?.let {
+                        tryToConnect(it)
+                    }
+                }
+    }
+
+    fun tryToConnect(config: JsonObject) {
+        val eventBusAddress = config.getString(CogboardConstants.Props.EVENT_ADDRESS)
+        try {
+            connect(config)
+        } catch (e: JSchException) {
+            vertx.eventBus().send(eventBusAddress, e)
+        }
+    }
+
+    fun connect(config: JsonObject) {
+        val user = config.getString(CogboardConstants.Props.USER) ?: ""
+        val pass = config.getString(CogboardConstants.Props.PASSWORD) ?: ""
+        val token = config.getString(CogboardConstants.Props.TOKEN) ?: ""
+        val key = config.getString(CogboardConstants.Props.SSH_KEY) ?: ""
+        val host = config.getString(CogboardConstants.Props.SSH_HOST) ?: ""
```

Comment:
To wygląda jakby można było to wrzucić w jedną klasę a przypisywanie tego zrobić w oddzielnej metodzie

Reply:
bardzo +1